framework/mongo/utils.py

```diff
@@ -63,10 +63,18 @@ def wrapper(cls):
 
 def get_or_http_error(Model, pk):
     instance = Model.load(pk)
+
     if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
+        if hasattr(instance, 'registrations'):
```

Comment:
It might make sense to allow the resource name to be passed in as an argument

```python
get_or_http_error(Node, 123, resource_name='project')
```

Or even the whole message

```python
get_or_http_error(Node, 123, message='This project has been deleted.')
```